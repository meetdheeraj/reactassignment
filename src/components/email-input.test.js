// vendors
import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from 'react-test-renderer';
// component
import EmailInput from "./email-input";

configure({ adapter: new Adapter() });

describe("Email Field", () => {
    it('renders correctly', () => {
        const rendered = renderer.create(
            <EmailInput value={"something@something.com"} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });

    xit("should return true for valid email with single @ and single dotshould render FormField with correct formTitle", () => {
        // test cases here
        const emailInput = shallow(<EmailInput value="qaz@g.com" />);

        expect(emailInput.value).toEqual('qaz@g.com');

        emailInput.find('input').simulate('onBlur');

        expect(emailInput.text()).toEqual('On');
    });

});
