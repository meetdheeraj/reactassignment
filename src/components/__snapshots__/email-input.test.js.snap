// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Field renders correctly 1`] = `
<input
  maxLength="255"
  onBlur={[Function]}
  spellCheck="false"
  tabIndex="3"
  type="email"
/>
`;
